### PR TITLE
Add an e2e workflow for the v0.35.x backport branch.

### PR DESCRIPTION
**Pull request by creachadair**

### CODE REVIEW

File: .github/workflows/e2e-nightly-35x.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      fail-fast: false
+      matrix:
+        p2p: ['legacy', 'new', 'hybrid']
+        group: ['00', '01', '02', '03']
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
```
Comment (williambanfield) [inline .github/workflows/e2e-nightly-35x.yml:26]: 1.17 :)

Comment (creachadair) [reply .github/workflows/e2e-nightly-35x.yml:26]: Oops, this wasn't supposed to be in this branch, haha.

Comment (creachadair) [reply .github/workflows/e2e-nightly-35x.yml:26]: Oh, this is not that PR. Yeah, I'll update once #7049 merges 🙂

Comment (creachadair) [reply .github/workflows/e2e-nightly-35x.yml]: Updated, PTAL. 😀

**Review by tychoish — APPROVED** [review]

**Review by williambanfield — APPROVED** [review]

**Pull request merged**